### PR TITLE
BF: define expected Series of correct for arch (eg i386) int

### LINKED ISSUE #14866: on 32bit -- Series is int64 but cut produces int32 series
#### Code Sample, a copy-pastable example if possible
failing on 32bit test
```python
    def test_single_bin(self):
        # issue 14652
        expected = Series([0, 0])

        s = Series([9., 9.])
        result = cut(s, 1, labels=False)
        tm.assert_series_equal(result, expected)

        s = Series([-9., -9.])
        result = cut(s, 1, labels=False)
        tm.assert_series_equal(result, expected)
```
#### Problem description
orig:  https://buildd.debian.org/status/fetch.php?pkg=pandas&arch=i386&ver=0.19.1%2Bgit174-g81a2f79-1&stamp=1481436461  (still don't know about "bad argument to internal function" -- can't replicate locally, so ignore for this one):
```
======================================================================
FAIL: test_single_bin (pandas.tools.tests.test_tile.TestCut)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«BUILDDIR»/pandas-0.19.1+git174-g81a2f79/debian/tmp/usr/lib/python2.7/dist-packages/pandas/tools/tests/test_tile.py", line 281, in test_single_bin
    tm.assert_series_equal(result, expected)
  File "/«BUILDDIR»/pandas-0.19.1+git174-g81a2f79/debian/tmp/usr/lib/python2.7/dist-packages/pandas/util/testing.py", line 1154, in assert_series_equal
    assert_attr_equal('dtype', left, right)
  File "/«BUILDDIR»/pandas-0.19.1+git174-g81a2f79/debian/tmp/usr/lib/python2.7/dist-packages/pandas/util/testing.py", line 878, in assert_attr_equal
    left_attr, right_attr)
  File "/«BUILDDIR»/pandas-0.19.1+git174-g81a2f79/debian/tmp/usr/lib/python2.7/dist-packages/pandas/util/testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: Attributes are different

Attribute "dtype" are different
[left]:  int32
[right]: int64
```

```
(Pdb) p expected
0    0
1    0
dtype: int64
(Pdb) p result
0    0
1    0
dtype: int32
(Pdb) p s
0    9.0
1    9.0
dtype: float64
```
so I guess if Series should use int64 regardless of underlying architecture default type, then cut should also produce int64, I guess.... or should Series have arch-default int??

### PR DESCRIPTION
**Pull request by yarikoptic**
- [x] Closes  #14866
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry (I believe it is nothing new ;) )

### CODE REVIEW

File: pandas/tools/tests/test_tile.py [hunk 1]
```diff
@@ -288,7 +288,10 @@ def test_qcut_duplicates_bin(self):
 
     def test_single_bin(self):
         # issue 14652
-        expected = Series([0, 0])
+        # Explicit dtype since Series produces int64 for ints, while cut
+        # (due to numpy.searchsorted) would use int32 on i386, so let's assure
+        # correct default to the architecture int
```
Comment (jreback) [inline pandas/tools/tests/test_tile.py]: we don't test on 32-bit, but if this works that is ok

specify ``dtype='intp'`` here

Comment (jreback) [inline pandas/tools/tests/test_tile.py]: though this works fine on windows now (e.g. 64-bit points on 32-bit systems).

and are you on master? this looks like an older line number.

Comment (yarikoptic) [inline pandas/tools/tests/test_tile.py]: why does it look old?  https://github.com/pandas-dev/pandas/blob/master/pandas/tools/tests/test_tile.py#L291

Comment (yarikoptic) [inline pandas/tools/tests/test_tile.py]: now with 'intp'

File: pandas/tests/indexes/common.py [hunk 2]
```diff
@@ -127,6 +127,13 @@ def test_ndarray_compat_properties(self):
 
         values = idx.values
         for prop in self._compat_props:
+            # skip equivalency check if converted type is object, as happens
+            # for PeriodIndex, since then object (address) would be 4 bytes
+            # on 32bit platforms and equivalence to int64 of original
+            # date time is just accidental
+            if prop in ('itemsize', 'nbytes') \
+                    and values.dtype.name == 'object':
+                continue
```
Comment (yarikoptic) [inline pandas/tests/indexes/common.py]: @jreback - I have added one more "skip" which was failing on i386 -- described "incode".  Does it make sense? ;)

File: pandas/tools/tests/test_tile.py (hunk 1 above)
Comment (jreback) [inline pandas/tools/tests/test_tile.py]: actually I think this is a bug, cut on ints should *always* be 64-bit regardless of arch.

File: pandas/tests/indexes/common.py (hunk 2 above)
Comment (jreback) [inline pandas/tests/indexes/common.py]: ok this is fine

File: pandas/tools/tests/test_tile.py (hunk 1 above)
Comment (jreback) [inline pandas/tools/tests/test_tile.py]: https://github.com/pandas-dev/pandas/blob/master/pandas/tools/tile.py#L218

needs an ``.astype('int64', copy=False)`` as these can get returned to the user in some paths; and they are 32-bit on 32-bit :< (while 64bit on others)

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
this was closed by https://github.com/pandas-dev/pandas/pull/15766

**Pull request closed without merging**